src/app/signing.service.ts

```diff
@@ -160,6 +160,38 @@ export class SigningService {
     return signedTransactionBytes.toString('hex');
   }
 
+  sign(seedHex: string, messageHex: string): string {
+    const privateKey = this.cryptoService.seedHexToPrivateKey(seedHex);
+
+    const contentBytes = new Buffer(messageHex, 'hex');
+    const contentLength = uvarint64ToBuf(contentBytes.length);
+
+    // We add a prefix to prevent the caller from signing a transaction
+    const prefix = 'DeSo Signed Message:\n';
+    const prefixBytes = new Buffer(prefix, 'hex');
```

Comment:
Ah so I got some bad news. You just got scammed by the Buffer library. You see we're living in a very degenerate world out there on the blockchain and we need to hand-verify all things. But worry not, I catch everything! So I was playing around with this, and turns out that while your algo is good overall, this lil `'hex'` guy is f-ing us. Try to sign something by hand and put `console.log(prefixBytes)` after this statement. Turns out `prefixBytes = [222]`! Not what we want. We would instead expect the `"DeSo Signed Message:\n"` string to turn into something like `[68 101 83 111 32 83 105 103 110 101 100 32 77 101 115 115 97 103 101 58 10]`. To do that, we need to change this into `new Buffer(prefix, 'utf-8')` just to make sure the Buffer dude knows we're passing an utf-8 string instead of a hex string.

Reply:
Fixed in 00aed39!